### PR TITLE
Add cdf function to InterRDF

### PR DESCRIPTION
**Pull request by VOD555**
Changes made in this Pull Request:
 - Add cdf function to InterRDF

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: pmda/test/test_rdf.py [hunk 1]
```diff
@@ -83,6 +83,15 @@ def test_same_result(sels, n_blocks):
     assert_almost_equal(nrdf.rdf, prdf.rdf)
 
 
+def test_cdf(sels):
+    s1, s2 = sels
+    rdf = InterRDF(s1, s2).run()
+    rdf.get_cdf()
+    cdf = np.cumsum(rdf.count) / rdf.nf
+    assert rdf.cdf[-1] == rdf.count.sum()/rdf.nf
```
Comment (orbeckst) [inline pmda/test/test_rdf.py]: Is this comparing floats or integers? If these are floats then you need to use `assert_almost_equal`.

File: CHANGELOG [hunk 2]
```diff
@@ -21,6 +21,7 @@ Enhancements
   * add timer for block-IO and block-compute
   * store block information in `_block` attribute (Issue #89)
   * add parallel density class (Issue #8)
+  * add `get_cdf` function to `pmda.rdf.InterRDF`
```
Comment (orbeckst) [inline CHANGELOG]: add `get_cdf` function to `pmda.rdf.InterRDF` to calculate cumulative distribution function

File: pmda/rdf.py [hunk 3]
```diff
@@ -161,6 +161,21 @@ def _conclude(self, ):
...
+        Note that this is the actual count within a given radius, i.e.,
+        :math:`N(r)`.
+
+        Returns
+        -------
+              cdf : numpy array
+                      a numpy array with the same structure as :attr:`rdf`
+        """
+        cdf = np.cumsum(self.count) / self.nf
+        self.cdf = cdf
```
Comment (orbeckst) [inline pmda/rdf.py]: With a `get` method this should not be stored. That's confusing (the expectation for such a function is to return a result) and it also uses up memory.

If you want to implement it as a managed attribute you could do
```python
@property
def cdf(self):
    return np.cumsum(self.count) / self.nf
```
Or calculate the cdf in `_conclude`, remove `get_cdf`, and just access `InterRDF.cdf`.

File: pmda/test/test_rdf.py [hunk 4]
```diff
@@ -83,6 +83,15 @@ def test_same_result(sels, n_blocks):
     assert_almost_equal(nrdf.rdf, prdf.rdf)
 
 
+def test_cdf(sels):
+    s1, s2 = sels
+    rdf = InterRDF(s1, s2).run()
+    rdf.get_cdf()
```
Comment (orbeckst) [inline pmda/test/test_rdf.py]: That's weird use of a `get_*` "function" – if anything, I'd expected
```python
cdf = rdf.get_cdf()
```
It's very confusing if then later magically an attribute `rdf.cdf` appears.

File: pmda/rdf.py (hunk 3 above)
Comment (VOD555) [reply pmda/rdf.py]: I prefer setting it as a property so that cdf will not be in memory when users don't want it.

Comment (VOD555) [reply pmda/rdf.py]: And I also changed the same function in InterRDF_s.

**Comment by orbeckst** [comment]
(And please fix PEP8...)

File: pmda/rdf.py (hunk 3 above)
Comment (orbeckst) [reply pmda/rdf.py]: ... and in MDA!

**Comment by orbeckst** [comment]
Please also address the remaining comments.

File: pmda/test/test_rdf.py (hunk 1 above)
Comment (VOD555) [reply pmda/test/test_rdf.py]: Fixed.

File: CHANGELOG [hunk 5]
```diff
@@ -21,6 +21,10 @@ Enhancements
   * add timer for block-IO and block-compute
   * store block information in `_block` attribute (Issue #89)
   * add parallel density class (Issue #8)
+  * add `cdf` attribute to `pmda.rdf.InterRDF` for cumulative
+    distribution function
+  * deprecate `get_cdf` method in `pmda.rdf.InterRDF_s` and use
```
Comment (orbeckst) [inline CHANGELOG]: Put this in a separate category *Changes*
```reST
* The method rdf.InterRDF_s.get_cdf() was removed; just use the managed attribute
  rdf.InterRDF_s.cdf
```

File: pmda/rdf.py [hunk 6]
```diff
@@ -318,7 +333,8 @@ def _conclude(self):
 
         self.rdf = rdf
 
-    def get_cdf(self):
+    @property
+    def cdf(self):
```
Comment (orbeckst) [inline pmda/rdf.py]: To the restructured text add (after *two* blank lines)
```reST

.. versionchanged:: 0.3.0
   Method ``get_cdf()`` was removed and :attr:`cdf` is now a managed attribute that computes
   the CDF when accessed.
```

File: CHANGELOG (hunk 5 above)
Comment (orbeckst) [reply CHANGELOG]: In MDAnalysis we would tag it with a deprecation and keep it around for a while. In PMDA we can move a bit faster.

File: pmda/rdf.py [hunk 7]
```diff
@@ -161,6 +161,21 @@ def _conclude(self, ):
...
+        """Calculate the cumulative distribution functions (CDF).
+        Note that this is the actual count within a given radius, i.e.,
+        :math:`N(r)`.
+
+        Returns
+        -------
+              cdf : numpy array
+                      a numpy array with the same structure as :attr:`rdf`
+        """
```
Comment (orbeckst) [inline pmda/rdf.py]: add a line (after *two* blank lines!)
```reST
.. versionadded:: 0.3.0
```

**Comment by VOD555** [comment]
@orbeckst All required changes have been done. Please have a look.

**Comment by orbeckst** [comment]
@VOD555 sorry, the PR fell off my radar. All looking good... will merge once Travis shows all green.

**Comment by orbeckst** [comment]
The RDF tests actually failed. Not sure why. Can you please look into this, @VOD555 ?

Perhaps changes in MDAnalysis?

**Comment by orbeckst** [comment]
Other tests (on master https://travis-ci.org/MDAnalysis/pmda ...) also fail in RDF, so it's not a problem with your PR but the reference values are not correct anymore. 

Please have a look and decide what to do: either adjust the references or raise an issue for the upstream change that breaks our tests here.

**Comment by orbeckst** [comment]
@VOD555 I opened #110 for the RDF failure.

**Comment by orbeckst** [comment]
Hopefully the lint test passes now, too.

**Comment by orbeckst** [comment]
I am rebasing to master and squashing commits locally. I will force push this branch with the rewritten history

**Pull request merged**